src/trimfastq.py

```diff
@@ -0,0 +1,247 @@
+            pass
+        else:
+            print('will include flowcell ID', flowcellID, 'in reads headers')
+
+    doRenameIDs = False
+    if '-renameIDs' in sys.argv:
+        doRenameIDs = True
+        RID = '@' + sys.argv[sys.argv.index('-renameIDs') + 1]
+
+    dotrim5=False
+    if '-trim5' in sys.argv:
+        dotrim5=True
+        trim5=int(sys.argv[sys.argv.index('-trim5')+1])
+        if doStdOut:
+            pass
+        else:
+            print('will trim ', trim5, 'bp from the 5-end')
+        outputfilename = inputfilename.split('.fastq')[0] + '.' +str(trim)+'bp-5prim-trim.fastq'
+
+    doAddEnd=False
+    if '-addEnd' in sys.argv:
+        doAddEnd=True
+        END=sys.argv[sys.argv.index('-addEnd')+1]
+        if doStdOut:
+            pass
+        else:
+            print('will add',  '/'+END, 'to read IDs')
+
+    doReplace=False
+    if '-replace' in sys.argv:
+        doReplace=True
+        oldstring=sys.argv[sys.argv.index('-replace')+1]
+        newstring=sys.argv[sys.argv.index('-replace')+2]
+        if newstring == 'blank':
+            newstring=''
+        if doStdOut:
+            pass
+        else:
+            print('will replace',  oldstring, 'with', newstring, 'in read IDs')
+
+    i=0 
+    shorter=0
+
+    if doStdOut:
+        pass
+    else:
+        outfile = open(outputfilename, 'w')
+
+    doStdIn = False
+    if inputfilename != '-':
+        if inputfilename.endswith('.bz2'):
+            cmd = 'bzip2 -cd ' + inputfilename
+        elif inputfilename.endswith('.gz'):
+            cmd = 'gunzip -c ' + inputfilename
+        else:
+            cmd = 'cat ' + inputfilename
+        p = os.popen(cmd, "r")
+    else:
+        doStdIn = True
+
+    line = 'line'
+
+    if dotrim5:
+        i=1
+        j=0
+        while line != '':
+            if doStdIn:
+                line = sys.stdin.readline()
+            else:
+                line = p.readline()
+            if line == '':
+                break
+            if i==1 and line[0]=='@':
+                if doFlowcellID and flowcellID not in line:
+                    ID='@'+flowcellID+'_'+line.replace(' ','_')[1:-1]+'\n'
+                else:
+                    ID=line.replace(' ','_')
+                if doReplace:
+                    ID=ID.replace(oldstring,newstring)
+                if doRenameIDs:
+                    ID = RID + str(j)
+                if doAddEnd:
+                    ID=ID.strip()+'/'+END+'\n'
+                i=2
+                continue
+            if i==2:
+                i=3
+                sequence=line[trim5:len(line)].strip()
+                continue
+            if i==3 and line[0]=='+':
+                plus='+\n'
+                i=4
+                continue
+            if i==4:
+                scores=line
+                i=1
+                scores=line[trim5:len(line)].strip()
+                scores=scores[0:trim]
+                j=j+1
+                if j % 5000000 == 0:
```

Comment:
Make a constant, whose name explains why this number is 5000000.

Reply:
done

Reply:
```
READS_LOG_INTERVAL=5000000
MILLION=1000000
```
Added these contants

Reply:
much clearer now.